### PR TITLE
Fix port level mTLS when Sidecar is defined

### PR DESCRIPTION
**Pull request by howardjohn**
This fixes one case in https://github.com/istio/istio/issues/27994 and captures the others which are still open issues

### CODE REVIEW

**Review by yangminzhu — COMMENTED** [review]
Thanks for the fix, I believe this is for case 5?

I have some local code for case 4 and can probably send out later once I verified it's fully working. 

I'm still not sure about case 3 and 6, It looks like if the port is explicitly omitted in sidecar.ingress, then it should be excluded from the passthrough filter chain in the first place (regardless of portLevelSetting)?

**Comment by howardjohn** [comment]
Yes, this solves only case 5 which I think we have consensus is a bug. The others are just extra tests capturing existing state, no changes there. I agree its a bit unclear what the expected behavior is. Lets add this to networking WG agenda

File: pilot/pkg/networking/core/v1alpha3/fake.go [hunk 1]
```diff
@@ -208,6 +208,9 @@ func (f *ConfigGenTest) SetupProxy(p *model.Proxy) *model.Proxy {
 	if p.ConfigNamespace == "" {
 		p.ConfigNamespace = "default"
 	}
+	if p.Metadata.Namespace == "" {
```
Comment (yangminzhu) [inline pilot/pkg/networking/core/v1alpha3/fake.go:211]: Is this related to the fix?

**Review by yangminzhu — APPROVED** [review]

File: pilot/pkg/networking/core/v1alpha3/fake.go (hunk 1 above)
Comment (howardjohn) [reply pilot/pkg/networking/core/v1alpha3/fake.go:211]: Yes, authn code looks for this and its not set in the test. This is just defaulting for test.

Maybe in the future we should make all code consistently use only ConfigNamespace and not Metadata.Namespace

**Review by ramaraochavali — APPROVED** [review]
